sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs

```diff
@@ -64,19 +65,84 @@ protected void PreInvoke(IExecutionContext executionContext)
                     continue;
                 }
 
-                if (scheme is AwsV4aAuthScheme || scheme is AwsV4AuthScheme)
+                executionContext.RequestContext.Signer = GetSigner(scheme);
+
+                if ((scheme is AwsV4aAuthScheme || scheme is AwsV4AuthScheme) && clientConfig.DefaultAWSCredentials != null)
                 {
                     // We can use DefaultAWSCredentials if it was set by the user for these schemes.
-                    executionContext.RequestContext.Identity = executionContext.RequestContext.ClientConfig.DefaultAWSCredentials;
+                    executionContext.RequestContext.Identity = clientConfig.DefaultAWSCredentials;
+                    break;
                 }
 
-                if (executionContext.RequestContext.Identity == null)
+#if NETFRAMEWORK
```

Comment:
I think we should add an else statement that calls the async version in a blocking way, similar what was done in `DefaultAWSTokenIdentityResolver.ResolveIdentity`, to account for the internal sync actions being called.

Reply:
Updated in https://github.com/aws/aws-sdk-net/pull/3562/commits/f2b2126357d7095d72c3cf889bb3c6411db788a9